src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java

```diff
@@ -0,0 +1,472 @@
+                    depth += (gt.hasExtendedAttribute(GATKVCFConstants.MIN_DP_FORMAT_KEY) ? Integer.parseInt((String)gt.getAnyAttribute(GATKVCFConstants.MIN_DP_FORMAT_KEY)) : (gt.hasDP() ? gt.getDP() : 0));
+                }
+            }
+
+            if ( loc.getStart() != vc.getStart() ) {
+                continue;
+            }
+
+            // special case ID (just preserve it)
+            if ( vc.hasID() ) {
+                rsIDs.add(vc.getID());
+            }
+
+            // add attributes
+            addReferenceConfidenceAttributes(vc.getAttributes(), annotationMap);
+        }
+
+        // when combining annotations use the median value from all input VCs which had annotations provided
+        for ( final Map.Entry<String, List<Comparable>> p : annotationMap.entrySet() ) {
+            if ( ! p.getValue().isEmpty() ) {
+                attributes.put(p.getKey(), combineAnnotationValues(p.getValue()));
+            }
+        }
+
+        if ( depth > 0 ) {
+            attributes.put(VCFConstants.DEPTH_KEY, String.valueOf(depth));
+        }
+
+        // remove stale AC and AF based attributes
+        removeStaleAttributesAfterMerge(attributes);
+
+        final String ID = rsIDs.isEmpty() ? VCFConstants.EMPTY_ID_FIELD : String.join(",", rsIDs);
+
+        // note that in order to calculate the end position, we need a list of alleles that doesn't include anything symbolic
+        final VariantContextBuilder builder = new VariantContextBuilder().source(name).id(ID).alleles(allelesList)
+                .chr(loc.getContig()).start(loc.getStart()).computeEndFromAlleles(nonSymbolicAlleles(allelesList), loc.getStart(), loc.getStart())
+                .genotypes(genotypes).unfiltered().attributes(new TreeMap<>(attributes)).log10PError(CommonInfo.NO_LOG10_PERROR);  // we will need to re-genotype later
+
+        return builder.make();
+    }
+
+    /**
+     * @param list  the original alleles list
+     * @return a non-null list of non-symbolic alleles
+     */
+    private static List<Allele> nonSymbolicAlleles(final List<Allele> list) {
+        final List<Allele> result = new ArrayList<>(list.size());
+        for ( final Allele allele : list ) {
+            if ( !allele.isSymbolic() )
+                result.add(allele);
+        }
+        return result;
+    }
+
+    /**
+     * Determines the ref allele given the provided reference base at this position
+     *
+     * @param VCs     collection of unsorted genomic VCs
+     * @param loc     the current location
+     * @param refBase the reference allele to use if all contexts in the VC are spanning
+     * @return new Allele or null if no reference allele/base is available
+     */
+    private static Allele determineReferenceAlleleGivenReferenceBase(final List<VariantContext> VCs, final Locatable loc, final Byte refBase) {
+        final Allele refAllele = GATKVariantContextUtils.determineReferenceAllele(VCs, loc);
+        if ( refAllele == null ) {
```

Comment:
simplify this logic - add one line `if (refAllele == null || refBase == null) { return null; }`